### PR TITLE
Minor tweaks

### PR DESCRIPTION
**Pull request by danielmarbach**
Few cleanups like

* Redundant, unnecessary code removed
* Unnecessary awaits removed
* Batch sending concurrently
* Proper await in OnError with CircuitBreaker

@Particular/azure-service-bus-maintainers please review

### CODE REVIEW

File: src/Transport/Seam/MessagePump.cs [hunk 1]
```diff
@@ -56,10 +56,10 @@ public Task Init(Func<PushContext, Task> pump, CriticalError criticalError, Push
         // For internal testing purposes.
         internal void OnError(Func<Exception, Task> func)
         {
-            topologyOperator.OnError(exception =>
+            topologyOperator.OnError(async exception =>
```
Comment (SeanFeldman) [inline src/Transport/Seam/MessagePump.cs]: The change failed the test and I've found an issue with CB thanks to that. Great :)

File: src/Transport/Sending/DefaultOutgoingBatchRouter.cs [hunk 2]
```diff
@@ -33,21 +33,24 @@ public DefaultOutgoingBatchRouter(IConvertOutgoingMessagesToBrokeredMessages out
...
+                pendingBatches.Add(RouteBatch(batch, context));
             }
+            return Task.WhenAll(pendingBatches);
         }
 
-        public async Task RouteBatch(Batch batch, ReceiveContext context)
+        public Task RouteBatch(Batch batch, ReceiveContext context)
         {
             var outgoingBatches = batch.Operations;
-            if (!outgoingBatches.Any()) return;
```
Comment (SeanFeldman) [inline src/Transport/Sending/DefaultOutgoingBatchRouter.cs]: I'm not entirely on board with this one. We remove the check for optimization. But we continue to execute the rest of the code. How much of an optimization is that anyway? Wouldn't exiting early and _not_ performing any other operations be better?

File: src/Transport/CircuitBreakers/RepeatedFailuresOverTimeCircuitBreaker.cs [hunk 3]
```diff
@@ -40,7 +40,7 @@ public Task Failure(Exception exception)
                 Logger.WarnFormat("The circuit breaker for {0} is now in the armed state", name);
             }
 
-            return Task.Delay(TimeSpan.FromSeconds(1));
+            return TaskEx.Completed;
```
Comment (danielmarbach) [inline src/Transport/CircuitBreakers/RepeatedFailuresOverTimeCircuitBreaker.cs]: Why don't you make the CB just sync then?

Comment (SeanFeldman) [inline src/Transport/CircuitBreakers/RepeatedFailuresOverTimeCircuitBreaker.cs]: Different PR. I've already made a mistake by fixing it here, which insight I shouldn't have done on this PR :)

**Pull request merged**